nncf/common/tensor_statistics/aggregator.py

```diff
@@ -33,8 +35,34 @@ class StatisticsAggregator(ABC):
     def __init__(self, dataset: Dataset):
         self.dataset = dataset
         self.stat_subset_size = None
+        self.batch_size = self.dataset.get_batch_size() or 1
+        dataset_len = self.dataset.get_length()
+        self.dataset_sample_size = dataset_len * self.batch_size if dataset_len is not None else dataset_len
         self.statistic_points = StatisticPointsContainer()
 
+    def _get_total_statistics_samples(
+        self,
+    ) -> Optional[int]:
+        """
+        Returns total number of statistics samples used.
+
+        :return: Total number of statistics samples used.
+        """
+        return (
+            min(self.dataset_sample_size or self.stat_subset_size, self.stat_subset_size)
+            if self.stat_subset_size is not None
+            else None
+        )
+
+    def _get_iterations_num(self, total_statistics_samples: Optional[int]) -> Optional[int]:
```

Comment:
Please, provide smth to support this statement

Reply:
See the `nncf/common/quantization/quantizers.py::get_num_levels` or `nncf/nncf/openvino/graph/node_utils.py::get_number_if_op`. They always provided numbers as the result.

Reply:
If the `get_length` returns `None`, that means that the dataset is infinite. That might happen.
What does mean when `get_batch_size` returns `None`?

Reply:
According to a description:
    def get_length(self) -> Optional[int]:
        """
        Tries to fetch length of the underlying dataset.
        :return: The length of the data_source if __len__() is implemented for it, and None otherwise.
        """
So `None` meaning that there is no a seeked method. It does not mean that the dataset contains inifinte elements

Reply:
So, it means `no implementation` for the basic `__len__()` method. What does `None` mean for `_get_iterations_num`?

Reply:
It is specified in docstring. 
`:return: The value of batch_size or _batch_size attributes of the data_source if exist, and None otherwise.`

Reply:
OkI got your point then I will reimplemet that part in the following way:
```
    def _get_iterations_num(self, total_statistics_samples: int) -> int:
        """
        Returns number of iterations to collect statistics.

        :param total_statistics_samples: Number of statistics samples are used.
        :return: Iterations number of statistics collection.
        """
        return total_statistics_samples // self.batch_size
```
and
```
iterataions_num = (self._get_iterations_num(calibration_samples_num) if calibration_samples_num is not None else None)
```

Reply:
I think we should review the correctness of the `None` as a value for `calibration_samples_num` as well.

Reply:
Sure. It is worth to create a new ticket. I will do